binaries/runtime/src/operator/python.rs

```diff
@@ -66,14 +66,24 @@ pub fn spawn(
             Python::with_gil(init_operator).wrap_err("failed to init python operator")?;
 
         while let Some(input) = inputs.blocking_recv() {
-            Python::with_gil(|py| {
+            let status_enum = Python::with_gil(|py| {
                 operator.call_method1(
                     py,
                     "on_input",
                     (input.id.to_string(), input.value, send_output.clone()),
                 )
             })
             .wrap_err("on_input failed")?;
+            let status_val = Python::with_gil(|py| status_enum.getattr(py, "value"))
+                .wrap_err("on_input must have enum return value")?;
+            let status: i32 = Python::with_gil(|py| status_val.extract(py))
+                .wrap_err("on_input has invalid return value")?;
+            match status {
+                0 => {}                                    // ok
+                1 => break,                                // stop
+                -1 => bail!("on_input returned an error"), // err
```

Comment:
I'm not sure if we should try to catch errors. By that, I mean that the only possible way to return a -1 is to catch the error and return -1. I think that this might make things hard to debug and go outside our scope.
We can try to catch errors outside of the operators, if we want a `not failing` mode, but that shouldn't be the default mode.

Reply:
I'm not completely sure what you mean, but the Python operator could do something like this:

```python
if http_response_code == 500:
    # internal server error -> exit with an error code
    return DoraStatus.ERR
```

So checking for the `-1` in the runtime makes sense to me...

Reply:
Ok, I don't say that the use case does not exist, but in the case you highlighted. It would have been better to raise an error directly in python than returning `-1` that will have no context as to why it failed.
I'm trying to say that we should probably try to catch native errors with native error handling rather than trying to implement our own error handling logic, which might at worst hide the underlying reason for the error.

Reply:
Yeah, for Python, native errors/exceptions make more sense. So we can remove it from the API if you like. We still need it for the shared library operators though because that's the only way to signal an error in C.

I guess the question is whether we want the operator API to be consistent across C/Rust/Python or whether we are okay with different forms of error handling. Personally, I'm fine with such minor differences since the APIs look different already (e.g. `Result` in Rust vs raw integers in C).

Reply:
Ok, I removed the error status for python operators in 128922ac65784191fee6ff7a2c8ecc1966d41913